### DISCUSSION TITLE
imodeljs-master\test-apps\ui-test-app

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#2036] [Q&A] [answered]
in readme.md from ui-test-app 
....
## Using the Edit frontstage to test editing an iModel
To test editing an iModel, the '.env' file in the ui-test-app directory should set the variable imjs_TESTAPP_ALLOW_WRITE=1. Optionally this variable can be set in the command shell before starting the servers. Files from iModelHub can be edited. Also, standalone files marked as editable can be edited using
the Electron application.

now i  send a revit file to iModelHub  by  iTwin Synchronizer ,i want set 
 how set Files from iModelHub can be edited？
 how standalone files marked as editable can be edited？
i want save the project geolocation point new value to my model 

![12312123123123123](https://user-images.githubusercontent.com/18022720/128998027-03390815-91d5-4bf5-a43f-7ef29c605f30.jpg)

**Comment by pmconne** [comment, marked as answer]
Editing standalone iModels is for testing purposes only - let's focus on editing briefcases (e.g., from iModelHub).
@bbastings can you explain the error @paprophet is seeing?

**Reply by bbastings** [reply]
@paprophet @pmconne The only case where the tool for "editor project location save" does not install without displaying a nicer error message is if IModelConnection.isReadonly returns false, ex. trying to edit a standalone imodel that does not support edit.

If this is a standalone imodel, you should have gotten a warning from ui-test-app when opening the model with imjs_TESTAPP_ALLOW_WRITE=1 set if the model doesn't allow edit.

* Also, it doesn't appear that the new lat/long values have been accepted yet (tool is prompting for accept point in view).

**Reply by pmconne** [reply]
Thanks @bbastings. @paprophet please open the iModel from iModelHub in ui-test-app.

**Reply by bbastings** [reply]
If this was already an iModel from iModelHub, and the problem is just that you haven't clicked in the view to accept your new lat/long, there should have been an additional toast messages (automatically closes after a short time). Open messages from the status bar to check.
![image](https://user-images.githubusercontent.com/65233531/129023995-80947c1f-071b-483f-9343-b35fa3f1b573.png)

**Reply by paprophet** [reply]
now when i open an iModel from iModelHub,show this error, 
there have some setting in iModelHub???

![editedit](https://user-images.githubusercontent.com/18022720/129160555-f85e6f01-d751-4c02-b20f-459a85633655.jpg)

**Reply by pmconne** [reply]
@bsteinbk

**Reply by bsteinbk** [reply]
@bbastings would know more about this, but from the error message `this.briefcaseConnection.txns` is undefined. Maybe Brien can elaborate on why/how `txns` would be or become undefined. If this is an expected situation we will need to update  `PushPullStatusField.tsx` in `ui-test-app` to test for that situation.

**Reply by pmconne** [reply]
@bsteinbk the problem is that you are lying about the kind of IModelConnection you opened.
```
  public static get briefcaseConnection(): BriefcaseConnection {
    return UiFramework.getIModelConnection()! as BriefcaseConnection;
  }
```

**Reply by bsteinbk** [reply]
@pmconne I did not write that code but I will look into updating it.

**Reply by pmconne** [reply]
This has been fixed. Closing as "answered".
